src/components/presentational/WeekCalendar/WeekCalendar.tsx

```diff
@@ -12,22 +13,23 @@ export interface WeekCalendarProps {
 	loading: boolean;
 	onStartDateChange?(startDate: Date): void;
 	dayRenderer(year: number, month: number, day: number, selectedWeek: boolean): JSX.Element | null;
+	innerRef?: React.Ref<HTMLDivElement>;
 }
 
 export default function (props: WeekCalendarProps) {
 	const element = useRef<HTMLDivElement>(null);
 
 	useEffect(() => {
 		if (props.onStartDateChange) {
-			var scrollListener = function (ev: Event) {
+			var scrollListener = debounce(function (ev: Event) {
```

Comment:
This debounce changed fixed a longstanding weird issue with this where it never really worked right on desktop due to weird orders of events; since essentially when it scrolls left to 0 we want it to immediate re-render and snap back to the middle scroll position...anyways, it works now